lib/krane/cli/run_command.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module Krane
+  module CLI
+    class RunCommand
+      DEFAULT_RUN_TIMEOUT = '300s'
+
+      OPTIONS = {
+        "global-timeout" => {
+          type: :string,
+          banner: "duration",
+          desc: "Timeout error is raised if the pod runs for longer than the specified number of seconds",
+          default: DEFAULT_RUN_TIMEOUT,
+        },
+        "arguments" => {
+          type: :string,
+          banner: '"ARG1 ARG2 ARG3"',
+          desc: "Override the default arguments for the command with a space-separated list of arguments",
+        },
+        "verify-result" => { type: :boolean, desc: "Wait for completion and verify pod success", default: true },
+        "command" => { type: :array, desc: "Override the default command in the container image" },
+        "template" => {
+          type: :string,
+          desc: "The template file you'll be rendering",
+          default: 'task-runner-template',
+        },
+        "env-vars" => {
+          type: :string,
+          banner: "VAR=val,FOO=bar",
+          desc: "A Comma-separated list of env vars",
+          default: '',
+        },
+      }
+
+      def self.from_options(namespace, context, options)
+        require "kubernetes-deploy/runner_task"
+        runner = KubernetesDeploy::RunnerTask.new(
+          namespace: namespace,
+          context: context,
+          max_watch_seconds: KubernetesDeploy::DurationParser.new(options["global-timeout"]).parse!.to_i,
```

Comment:
I did this too but we probably shouldn't have a bad arg causing a stack dump. We can defer fixing this to the refactor you're planning. 

```
➜  kubernetes-deploy git:(add-krane-run-command) ✗ be krane run blah blah --global-timeout 3
bundler: failed to load command: krane (/Users/danielturner/.gem/ruby/2.3.3/bin/krane)
KubernetesDeploy::DurationParser::ParsingError: Invalid ISO 8601 duration: "3"
  /Users/danielturner/src/github.com/Shopify/kubernetes-deploy/lib/kubernetes-deploy/duration_parser.rb:25:in `rescue in rescue in parse!'
  /Users/danielturner/src/github.com/Shopify/kubernetes-deploy/lib/kubernetes-deploy/duration_parser.rb:22:in `rescue in parse!'
  /Users/danielturner/src/github.com/Shopify/kubernetes-deploy/lib/kubernetes-deploy/duration_parser.rb:20:in `parse!'
```

Reply:
😬 I'll make that a part of my next PR for sure.